### PR TITLE
Extraction of browser history

### LINKED ISSUE #178: Extraction of browser history
**Is your feature request related to a problem? Please describe.**
Extraction of browser history.
Implementation from the command line to extract browser history directly.
Creating API which will give the extracted browser history data(this will help in frontend).

### PR DESCRIPTION
**Pull request by swapnalshahil**
# Description

Extraction of browser history.
Implementation from the command line to extract browser history directly.
Creating API which will give the extracted browser history data(this will help in frontend).
API: /data/<case_name>/browsers

Fixes #178 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

POSTMAN and CMD

USING CMD
2021-06-17_LI
2021-06-17 (1)_LI

Extraction from POSTMAN (above)

history.tsv file (above)

API (above)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by swapnalshahil** [comment]
@shivanshu1333 sir! Please have a look :)

File: flask-backend/api/routes/data.py [hunk 1]
```diff
@@ -17,6 +16,14 @@
...
     File = cases_data_path +  case_name + "/tsv/" + "sms.tsv"
     os.chdir(ROOT_DIR)
     return File
+
+
+@data.route('/<case_name>/browsers',methods = ["GET"])
+def browsers(case_name):
+    File = cases_data_path + case_name + "/tsv/" + "history.tsv"
+    os.chdir(ROOT_DIR)
+    return File
```
Comment (sachinsom93) [inline flask-backend/api/routes/data.py:29]: Hey @swapnalshahil, will these data APIs work in case of duplicate case names ?

**Review by sachinsom93 — DISMISSED** [review]

File: scripts/browser.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    print("\n" + str(len(browser_dict.values())) +
+          " browser history were processed")
+
+
+def store_browser_history_data(session_name):
+    global browserdb
+    global OUTPUT
+    OUTPUT = OUTPUT + SEP + 'data' + SEP + session_name
+    browserdb = OUTPUT + SEP + 'db/History'
```
Comment (shivanshuraj1333) [inline scripts/browser.py:121]: Are we able to handle the case where we have multiple browsers installed in our device, and we want to extract all the browser history?
Do we have different db locations for other browsers or a common db shared by different browsers? If they all share the same db then extracting details of that db will work fine, but if all the browsers contain different dbs, we need to extract data from all the dbs.

**Comment by swapnalshahil** [comment]
@shivanshu1333  sir! solved merge conflict. You may review!🙂

File: scripts/browser.py (hunk 2 above)
Comment (swapnalshahil) [reply scripts/browser.py:121]: Yes sir! Browsers like Mozilla Firefox, opera have different dbs. As you said I will add those dbs. But now we can only extract chrome and the default browser.

Comment (swapnalshahil) [reply scripts/browser.py:121]: @shivanshu1333 sir! Added Mozilla and opera dbs. While extracting dbs of these browsers will be available in /data/db/

Comment (swapnalshahil) [reply scripts/browser.py:121]: Also sir there are lots of browsers in market. And it will be hard to add all those dbs. 😅

Comment (swapnalshahil) [reply scripts/browser.py:121]: Also sir! there are so many browsers in the market it is nearly impossible to add all dbs. So it's better to extract dbs of mostly used browsers only.

Comment (shivanshuraj1333) [reply scripts/browser.py:121]: Extracting history from default browser and the common browsers will work

**Review by shivanshuraj1333 — APPROVED** [review]
LGTM

**Pull request merged**